charms/katib-db-manager/src/charm.py

```diff
@@ -165,123 +168,162 @@ def _on_mysql_relation(self, event):
         try:
             relation = self.model.get_relation("relational-db")
             if relation:
-                self.logger.warnning(
-                    "Failed to create mysql relation due to existing relational-db relation."
+                self.logger.warning(
+                    "Up-to-date database relation relational-db is already established."
+                )
+                self.logger.warning(f"{MYSQL_WARNING} {UNBLOCK_MESSAGE}")
+                self.model.unit.status = BlockedStatus(
+                    f"{MYSQL_WARNING} {UNBLOCK_MESSAGE} See logs"
                 )
                 return
         except KeyError:
-            # relational-db relation was not found, proceed
             pass
+        # relational-db relation was not found, proceed with warnings
+        self.logger.warning(MYSQL_WARNING)
+        self.model.unit.status = MaintenanceStatus(f"Adding mysql relation. {MYSQL_WARNING}")
         self._on_event(event)
 
-    def _on_mysql_relation_removed(self, event):
-        """Remove mysql relation data."""
-        pass
+    def _on_mysql_relation_remove(self, event):
+        """Process removal of mysql relation."""
+        self.model.unit.status = MaintenanceStatus(f"Removing mysql relation. {MYSQL_WARNING}")
+        self._on_event(event)
 
     def _on_relational_db_relation(self, event):
         """Check for existing database relations and process relational-db relation if needed."""
-        # check for mysql relation
         # relying on KeyError to ensure that mysql relation is not present
-        assert 0
         try:
             relation = self.model.get_relation("mysql")
             if relation:
-                self.logger.warninig(
-                    "Failed to create relational-db relation due to existing mysql relation"
+                self.logger.warning(
+                    "Failed to create relational-db relation due to existing mysql relation."
+                )
+                self.logger.warning(f"{MYSQL_WARNING} {UNBLOCK_MESSAGE}")
+                self.model.unit.status = BlockedStatus(
+                    f"{MYSQL_WARNING} {UNBLOCK_MESSAGE} See logs"
```

Comment:
I really think we need to do something with messaging. 

eg
"mysql and relational-db cannot co-exist. `mysql` is deprecated, please remove it"

in logs we can go extremely explicit.

in status need to play to see the limit. In the worst case "Relations conflict, please see logs"